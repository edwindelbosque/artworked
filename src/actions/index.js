export const setResults = results => ({
	type: 'SET_RESULTS',
	results
});

export const toggleLoading = boolean => ({
	type: 'TOGGLE_LOADING',
	boolean
});

export const addFavorite = item => ({
	type: 'ADD_FAVORITE',
	item
});

export const removeFavorite = id => ({
	type: 'REMOVE_FAVORITE',
	id
});
