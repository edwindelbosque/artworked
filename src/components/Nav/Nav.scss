.Nav {
	display: flex;
	justify-content: center;
	width: 100%;
	position: fixed;
	background: rgba(0, 0, 0, 0.815);
	backdrop-filter: blur(10px);
	z-index: 1;
	border-bottom: 1px solid rgb(163, 180, 255);
	top: 0;

	h2 {
		font-size: 36px;
		letter-spacing: 4px;
		font-family: 'Text Me One', sans-serif;
		color: #ffffff;
		padding: 4px 0;
		width: 100%;
		text-align: center;
	}
}
