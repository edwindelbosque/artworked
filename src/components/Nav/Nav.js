import React from 'react';
import './Nav.scss';
import { connect } from 'react-redux';
import { toggleFavorites } from '../../actions/index';
import { bindActionCreators } from 'redux';
import { Link, Route } from 'react-router-dom';

const Nav = ({ isFavorites, toggleFavorites }) => {
	const handleClick = e => {
		toggleFavorites();
	};

	const toggleStyling = isFavorites ? 'active' : '';

	return (
		<nav className='Nav'>
			<Route exact path='/(|favorites)'>
				<div className='container'>
					<h3>My List</h3>
					<div className={`slider-track ${toggleStyling}`}>
						<Link to={isFavorites ? '/' : 'favorites'}>
							<div className='slider' onClick={handleClick}></div>
						</Link>
					</div>
				</div>
			</Route>
			<h2>Artworked</h2>
			<div></div>
		</nav>
	);
};

const mapStateToProps = ({ isFavorites }) => ({
	isFavorites
});

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ toggleFavorites }, dispatch);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Nav);
