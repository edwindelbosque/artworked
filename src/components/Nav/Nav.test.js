import React from 'react';
import { Nav, mapDispatchToProps, mapStateToProps } from './Nav';
import { toggleFavorites } from '../../actions/index';
import { shallow } from 'enzyme';

describe('Nav', () => {
	let wrapper;
	const toggleFavorite = jest.fn();
	const isFavorites = false;

	beforeEach(() => {
		wrapper = shallow(
			<Nav toggleFavorites={toggleFavorite} isFavorites={isFavorites} />
		);
	});

	it('should match Nav snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should invoke handleClick on button click', () => {
		wrapper.find('.slider').simulate('click');
		expect(toggleFavorite).toHaveBeenCalled();
	});

	describe('mapDispatchToProps', () => {
		const mockDispatch = jest.fn();

		it('should call dispatch with a toggleFavorites action when handleClick is called', () => {
			const actionToDispatch = toggleFavorites();

			const mappedProps = mapDispatchToProps(mockDispatch);
			mappedProps.toggleFavorites();

			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
		});
	});

	describe('mockStateToProps', () => {
		it('should return a boolean', () => {
			const mockState = {
				isFavorites: false
			};
			const expected = {
				isFavorites: false
			};

			const mappedProps = mapStateToProps(mockState);

			expect(mappedProps).toEqual(expected);
		});
	});
});
