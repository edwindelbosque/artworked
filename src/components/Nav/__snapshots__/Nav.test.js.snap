// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match Nav snapshot 1`] = `
<nav
  className="Nav"
>
  <Route
    exact={true}
    path="/(|favorites)"
  >
    <div
      className="container"
    >
      <h3>
        My List
      </h3>
      <div
        className="slider-track "
      >
        <Link
          to="favorites"
        >
          <div
            className="slider"
            onClick={[MockFunction]}
          />
        </Link>
      </div>
    </div>
  </Route>
  <h2>
    Artworked
  </h2>
  <div />
</nav>
`;
